api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -124,18 +107,14 @@
   public ResponseEntity<DataSet> createDataSet(
       String workspaceNamespace, String workspaceFirecloudName, DataSetRequest dataSetRequest) {
     validateDataSetCreateRequest(dataSetRequest);
-    final Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
-        workspaceNamespace, workspaceFirecloudName, WorkspaceAccessLevel.WRITER);
     final long workspaceId =
-        workspaceService.get(workspaceNamespace, workspaceFirecloudName).getWorkspaceId();
+        workspaceService
+            .getWorkspaceEnforceAccessLevelAndSetCdrVersion(
+                workspaceNamespace, workspaceFirecloudName, WorkspaceAccessLevel.WRITER)
+            .getWorkspaceId();
     dataSetRequest.setWorkspaceId(workspaceId);
-    DbDataset datasetToSave = dataSetMapper.dataSetRequestToDb(dataSetRequest, null);
-    datasetToSave.setCreationTime(now);
-    datasetToSave.setCreatorId(userProvider.get().getUserId());
-    datasetToSave.setInvalid(false);
-    DbDataset savedDataSet = dataSetService.saveDataSet(datasetToSave);
-    return ResponseEntity.ok(dataSetMapper.dbModelToClient(savedDataSet));
+    return ResponseEntity.ok(
+        dataSetService.saveDataSet(dataSetRequest, userProvider.get().getUserId()));
```

Comment:
Not part of this story, but we should add validation for createRequest to be sure we have cohort/conceptsets and values information in data set Request

Reply:
Isn't that what `validateDataSetCreateRequest(dataSetRequest)` on line 109 does?

Reply:
Ahh i missed that. Thanks!